src/hooks/useFetchHiddenTokens.ts

```diff
@@ -1,27 +1,61 @@
 import { useQuery } from '@tanstack/react-query';
-import useAccountSettings from './useAccountSettings';
-import { getHiddenTokens } from '@/handlers/localstorage/accountLocal';
 import { getPreference } from '@/model/preferences';
+import { time } from '@/utils';
+import { useOpenCollectionsStore } from '@/state/nfts/openCollectionsStore';
+import { isDataComplete } from '@/state/nfts/utils';
+import { useNftsStore } from '@/state/nfts/nfts';
 
-export const hiddenTokensQueryKey = ({ address }: { address?: string }) => ['hidden-tokens', address];
-
-export default function useFetchHiddenTokens({ address }: { address?: string }) {
-  const { network } = useAccountSettings();
-
-  return useQuery(
-    hiddenTokensQueryKey({ address }),
-    async () => {
-      if (!address) return [];
-      let hiddenTokens = await getHiddenTokens(address, network);
-      const hiddenTokensFromCloud = await getPreference('hidden', address);
-      if (hiddenTokensFromCloud?.hidden?.ids && hiddenTokensFromCloud?.hidden?.ids.length > 0) {
-        hiddenTokens = hiddenTokensFromCloud.hidden.ids;
-      }
-
-      return hiddenTokens as string[];
-    },
-    {
-      enabled: Boolean(address),
+export const hiddenTokensQueryKey = ({ address }: { address: string }) => ['hidden-tokens', address];
+
+const STABLE_ARRAY: string[] = [];
+
+export async function getHidden(address: string, isMigration = false) {
+  if (!address) return STABLE_ARRAY;
+
+  const hiddenTokens = await getPreference('hidden', address);
+  if (hiddenTokens?.hidden?.ids?.length) {
+    const tokens = hiddenTokens.hidden.ids;
+    if (!isDataComplete(tokens) && !isMigration) {
+      useOpenCollectionsStore.getState(address).setCollectionOpen('hidden', false);
+
+      const { fetch } = useNftsStore.getState(address);
+
+      const data = await fetch(
+        { collectionId: 'hidden', isMigration: true },
+        {
+          force: true,
+          updateQueryKey: false,
+          cacheTime: time.infinity,
+          staleTime: time.infinity,
+        }
+      );
+
+      if (!data) return tokens;
+
+      useNftsStore.setState(state => {
+        const now = Date.now();
+        return {
+          nftsByCollection: new Map([...state.nftsByCollection, ...data.nftsByCollection]),
```

Comment:
i thought you had to do ...state on the return or else it wipes away the rest of the state?